mayavi/tests/test_garbage_collection.py

```diff
@@ -0,0 +1,110 @@
+""" Tests for the garbage collection of objects in mayavi package.
+"""
+# Authors: Deepak Surti, Ioannis Tziakos
+# Copyright (c) 2015, Enthought, Inc.
+# License: BSD Style.
+
+from numpy import sqrt, sin, mgrid
+from traits.api import Instance, HasTraits
+from traitsui.api import View, Item
+from tvtk.pyface.scene_editor import SceneEditor
+from tvtk.tests.common import TestGarbageCollection
+from mayavi.core.ui.mayavi_scene import MayaviScene
+from mayavi.core.engine import Engine
+from mayavi.core.null_engine import NullEngine
+from mayavi.tools.mlab_scene_model import MlabSceneModel
+
+class TestMayaviGarbageCollection(TestGarbageCollection):
+    """ See: tvtk.tests.common.TestGarbageCollection
+    """
+    def test_mlab_scene_model_with_gui(self):
+        """ Tests if MlabSceneModel with GUI can be garbage collected."""
+        class MlabApp(HasTraits):
+
+            # The scene model.
+            scene = Instance(MlabSceneModel, ())
+
+            view = View(Item(name='scene',
+                             editor=SceneEditor(scene_class=MayaviScene),
+                             show_label=False,
+                             resizable=True,
+                             height=500,
+                             width=500),
+                        resizable=True
+                        )
+
+            def __init__(self, **traits):
+                self.generate_data()
+
+            def generate_data(self):
+                # Create some data
+                X, Y = mgrid[-2:2:100j, -2:2:100j]
+                R = 10*sqrt(X**2 + Y**2)
+                Z = sin(R)/R
+                self.scene.mlab.surf(X, Y, Z, colormap='gist_earth')
+
+        def create_fn():
+            app = MlabApp()
+            return app.edit_traits()
+
+        self.check_object_garbage_collected(create_fn)
```

Comment:
I think that you need to use another dummy class with `create and destroy` methods.

```python
class Dummy(object):

    def create(self):
          self.app = app = MlabApp()
          self.ui = app.edit_traits()
          return app.scene

    def destroy(self):
          self.ui.dispose()
          self.ui = None
          self.app = None
```

To make sure that dispose is called